pymeasure/display/widgets/table_widget.py

```diff
@@ -0,0 +1,546 @@
+        return pixelmap
+
+
+class Table(QtWidgets.QTableView):
+    """ Table format view of :class:`Experiment<pymeasure.display.manager.Experiment>`
+    objects
+
+    """
+
+    supported_formats = {
+        'csv': "CSV file (*.csv)",
+        'excel': "Excel file (*.xlsx)",
+        'html': "HTML file (*.html *.htm)",
+        'json': "JSON file (*.json)",
+        'latex': "LaTeX file (*.tex)",
+        'markdown': "Markdown file (*.md)",
+        'xml': "XML file (*.xml)",
+    }
+
+    def __init__(self, refresh_time=0.2, check_status=True,
+                 force_reload=False, by_column=True, parent=None):
+        super().__init__(parent)
+        self.force_reload = force_reload
+        if by_column:
+            model = PandasModelByColumn()
+        else:
+            model = PandasModelByRow()
+
+        if SORTING_ENABLED:
+            proxyModel = QtCore.QSortFilterProxyModel(self)
+            proxyModel.setSourceModel(model)
+            model = proxyModel
+
+            model.setSortRole(SORT_ROLE)
+        self.setModel(model)
+        self.horizontalHeader().setStyleSheet("font: bold;")
+        self.setSortingEnabled(True)
+        self.sortByColumn(-1, QtCore.Qt.SortOrder.AscendingOrder)
+        self.horizontalHeader().setSectionsMovable(True)
+        self.horizontalHeader().setSectionResizeMode(
+            QtWidgets.QHeaderView.ResizeMode.ResizeToContents
+        )
+
+        self.setup_context_menu()
+
+        self.refresh_time = refresh_time
+        self.check_status = check_status
+        self.timer = QtCore.QTimer()
+        self.timer.timeout.connect(self.update_tables)
+        self.timer.start(int(self.refresh_time * 1e3))
+
+    def composed_dataframe(self):
+        """ Create single pandas dataframe out of the dataframe list """
+        if SORTING_ENABLED:
+            model = self.model().sourceModel()
+        else:
+            model = self.model()
+
+        df_list = [result.data for result in model.results_list]
+        if isinstance(model, PandasModelByRow):
+            # Concatenate pandas data frames
+            df = pd.concat(df_list, axis=0).replace(to_replace=NaN, value="")
+        else:
+            # Concatenate pandas data frames
+            df = pd.concat(df_list, axis=1).replace(to_replace=NaN, value="")
+
+        return df
+
+    def export_action(self):
+        df = self.composed_dataframe()
+
+        formats = ";;".join(self.supported_formats.values())
+        filename_and_ext = QtWidgets.QFileDialog.getSaveFileName(self,
+                                                                 "Save File",
+                                                                 "",
+                                                                 formats)
+        filename = filename_and_ext[0]
+        ext = filename_and_ext[1]
+        if filename:
+            for k, v in self.supported_formats.items():
+                if v == ext:
+                    break
```

Comment:
Minor thingy: seeing that you only use this to populate the formats of the FileDialog (a bit above) and here, maybe it is easier to swap the keys and values of the `supported_formats` dict, such that you can do a normal lookup over here.

(I do find it rather annoying that this loop is one of the best ways to get a key from a value in a dict....)